fvm/environment/contract_reader.go

```diff
@@ -69,12 +72,37 @@ func (reader *ContractReader) ResolveLocation(
 		return nil, fmt.Errorf("resolve location failed: %w", err)
 	}
 
+	return ResolveLocation(
+		identifiers,
+		location,
+		reader.accounts.GetContractNames,
+		reader.cryptoContractAddress,
+	)
+}
+
+func ResolveLocation(
+	identifiers []ast.Identifier,
+	location common.Location,
+	getContractNames func(flow.Address) ([]string, error),
+	cryptoContractAddress common.Address,
+) ([]runtime.ResolvedLocation, error) {
+
 	addressLocation, isAddress := location.(common.AddressLocation)
 
 	// if the location is not an address location, e.g. an identifier location
-	// (`import Crypto`), then return a single resolved location which declares
-	// all identifiers.
+	// then return a single resolved location which declares all identifiers.
 	if !isAddress {
+
+		// if the location is the Crypto contract,
+		// translate it to the address of the Crypto contract on the chain
+
```

Comment:
Does it matter to us if the user imports it directly (by using the address) or indirectly (via this functionality). If it does mater maybe we meter this call, so that we have an option to change the weight later.

Reply:
No, it should not matter. Users need to keep being able to import using the identifier location (`import Crypto`), and if they import from the address that is also OK.

Not really sure why we would need extra metering code? 🤔